finagle-mysql/src/main/scala/com/twitter/finagle/mysql/CanBeParameter.scala

```diff
@@ -0,0 +1,182 @@
+package com.twitter.finagle.exp.mysql
+
+import com.twitter.finagle.exp.mysql.transport.{Buffer, BufferWriter}
+
+trait CanBeParameter[-A] { outer =>
+  /**
+   * Returns the size of the given parameter in its MySQL binary representation.
+   **/
+  def sizeOf(param: A): Int
+
+  /**
+   * Retrieves the MySQL type code for the given parameter.
+   * */
+  def typeCode(param: A): Short
+
+  def write(writer: BufferWriter, param: A): Unit
```

Comment:
@mosesn That seems to give a wrong impression that this is a pure function, which it is not. It's executed purely for its side-effects. And if it's just going to return the same writer at the end, the return doesn't mean much. If a function is side-effecting it should show in its type-signature, don't you think?